chill-algebird/src/main/scala/com/twitter/chill/algebird/AlgebirdSerializers.scala

```diff
@@ -95,3 +86,42 @@ class HLLMonoidSerializer extends KSerializer[HyperLogLogMonoid] {
     hllMonoids.getOrElseUpdate(bits, new HyperLogLogMonoid(bits))
   }
 }
+
+class SketchMapSerializer[K, V](implicit skmMonoid: SketchMapMonoid[K, V], valueMonoid: Monoid[V])
+  extends KSerializer[SketchMap[K, V]] {
+
+  def write(kryo: Kryo, output: Output, skm: SketchMap[K, V]) {
+    val rows = skm.valuesTable.rows
+    val cols = skm.valuesTable.cols
+
+    val values: Seq[((Int, Int), V)] =
+      for (
+        r: Int <- 0 until rows; c: Int <- 0 until cols if valueMonoid.isNonZero(skm.valuesTable.getValue((r, c)))
+      ) yield ((r, c), skm.valuesTable.getValue((r, c)))
+
+    output.writeInt(rows, true)
+    output.writeInt(cols, true)
+    kryo.writeClassAndObject(output, values)
+    kryo.writeClassAndObject(output, skm.totalValue)
+    kryo.writeClassAndObject(output, skm.heavyHitterKeys)
+
+    output.flush()
+  }
+
+  def read(kryo: Kryo, in: Input, cls: Class[SketchMap[K, V]]): SketchMap[K, V] = {
+    val rowsOrig = in.readInt(true)
+    val colsOrig = in.readInt(true)
+    val values = kryo.readClassAndObject(in).asInstanceOf[Seq[((Int, Int), V)]]
+    val totalValue = kryo.readClassAndObject(in).asInstanceOf[V]
+    val heavyHitterKeys = kryo.readClassAndObject(in).asInstanceOf[List[K]]
+
+    val rows = if (rowsOrig == 0) skmMonoid.params.depth else rowsOrig
```

Comment:
I am not 100% sure what lines you refer to, but would assume that 
` val rows = if (rowsOrig == 0) skmMonoid.params.depth else rowsOrig` and `val cols = if (colsOrig == 0) skmMonoid.params.width else colsOrig`.
 As I was saying that we wrote this serializer when we were using algebird 0.4 and in that version the implementation of `SketchMap` had a bug/feature that when you serialize empty `SketchMap` the `val rows = skm.valuesTable.rows` and `val cols = skm.valuesTable.cols` (see "write" method) would be set to 0 and to the size of the underneath supporting AdaptiveMatrix. Since this PR is for version of chill which is using algebird 0.10 these lines can be removed (I would prefer to still have them, so we do not need to recalculate our old SkatchMaps, but if it is a deal breaker, sure I can remove them).